application/controllers/ContactGroupsController.php

```diff
@@ -0,0 +1,184 @@
+                $sortControl->getSortParam()
+            ]
+        );
+
+        if ($searchBar->hasBeenSent() && ! $searchBar->isValid()) {
+            if ($searchBar->hasBeenSubmitted()) {
+                $filter = $this->getFilter();
+            } else {
+                $this->addControl($searchBar);
+                $this->sendMultipartUpdate();
+
+                return;
+            }
+        } else {
+            $filter = $searchBar->getFilter();
+        }
+
+        $groups->filter($filter);
+
+        $this->addControl($paginationControl);
+        $this->addControl($sortControl);
+        $this->addControl($limitControl);
+        $this->addControl($searchBar);
+        $this->addContent(
+            (new ButtonLink(
+                Text::create(t('Add Contact Group')),
+                Links::contactGroupsAdd(),
+                'plus',
+                ['class' => 'add-new-component']
+            ))->openInModal()
+        );
+
+        $this->addContent(new ContactGroupList($groups));
+
+        if (! $searchBar->hasBeenSubmitted() && $searchBar->hasBeenSent()) {
+            $this->sendMultipartUpdate();
+        }
+
+        $this->setTitle(t('Contact Groups'));
+        $this->getTabs()->activate('contact-groups');
+    }
+
+    public function addAction(): void
+    {
+        $form = (new ContactGroupForm(Database::get()))
+            ->setAction((string)Links::contactGroupsAdd())
+            ->on(Form::ON_SENT, function (ContactGroupForm $form) {
+                if (! $form->hasBeenSubmitted()) {
+                    foreach ($form->getPartUpdates() as $update) {
+                        if (! is_array($update)) {
+                            $update = [$update];
+                        }
+
+                        $this->addPart(...$update);
+                    }
+                }
+            })
+            ->on(Form::ON_SUCCESS, function (ContactGroupForm $form) {
+                $groupIdentifier = $form->addGroup();
+                if ($groupIdentifier) {
+                    Notification::success(t('New group has been successfully added'));
+                    $this->sendExtraUpdates(['#col1']);
+                    $this->getResponse()->setHeader('X-Icinga-Container', 'col2');
+                    $this->redirectNow(Links::contactGroup($groupIdentifier));
+                } else {
+                    Notification::error(t('Failed to add new contact group'));
+                }
+            })
+            ->handleRequest($this->getServerRequest());
+
+        $this->addContent($form);
+        $this->setTitle(t('Add Contact Group'));
+    }
+
+    public function suggestMemberAction(): void
+    {
+        $members = new MemberSuggestions();
+        $members->forRequest($this->getServerRequest());
+
+        $this->getDocument()->addHtml($members);
+    }
+
+    public function getTabs(): Tabs
+    {
+        if ($this->getRequest()->getActionName() === 'index') {
```

Comment:
Why is this necessary? All other actions don't show tabs, so I don't see a reason for this